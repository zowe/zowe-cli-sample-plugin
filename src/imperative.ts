/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import {IImperativeConfig} from "@brightside/imperative";

const config: IImperativeConfig = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    overrides: {
        CredentialManager: "./overrides/CredentialManager.override"
    },
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI sample plug-in",
    pluginAliases: ["zcsp"],
    rootCommandDescription: "Welcome to the sample plug-in for Zowe CLI!\n\n The sample plug-in " +
        "(& CLI) follows the Zowe CLI command syntax 'zowe [group] [action] [object] [options]'. " +
        "Where, in the case of the plugin, " +
        "the [group] is the package.json name, " +
        "the [actions] are defined in the project under 'src/cli/', " +
        "& the [objects] (definitions & handler) are defined in the project under 'src/cli/'.",
    productDisplayName: "Zowe CLI Sample Plug-in",
    name: "zowe-cli-sample"
};

export = config;
