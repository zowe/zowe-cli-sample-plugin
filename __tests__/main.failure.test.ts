/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { Imperative } from "@brightside/imperative";
// The main tests exist in seperate files because of module resolution in Jest,
// we want to require the main module a few times to perform different tests
describe("main failure", () => {
    it("should log an imperative init failure", (done) => {
        let logMsgs: string = "";

        // Mock imperative init
        Imperative.init = jest.fn((args): Promise<void> => {
            return new Promise<void>((fulfill, reject) => {
                reject("Imperative init failed!");
            });
        });

        // Mock the console error
        Imperative.console.error = jest.fn((msg) => {
            logMsgs += msg;
            expect(logMsgs).toMatchSnapshot();
            done();
        });

        // Require should run the init
        const main = require("../src/main");
    });
});
