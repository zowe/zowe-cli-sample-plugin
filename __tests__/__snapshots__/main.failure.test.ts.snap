// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main failure should log an imperative init failure 1`] = `
"An error occured during imperative initalization:
Imperative init failed!"
`;
